src/app/courses/[...courseId]/page.tsx

```diff
@@ -20,7 +20,24 @@ const checkAccess = async (courseId: string) => {
     return false;
   }
   const purchases = await getPurchases(session.user.email);
-  if (purchases.map((p) => p.id).includes(Number(courseId))) {
+  if (
+    purchases
+      .map(
```

Comment:
Can you create an interface out of this?
You might be able to get this type from prisma

Reply:
Sure! on it